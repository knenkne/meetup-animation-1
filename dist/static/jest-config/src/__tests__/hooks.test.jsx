import React from 'react'
import { mount } from 'enzyme'

import { ComponentWithCallback, ComponentWithHook } from '../component-with-hook'

describe('Hook tests', () => {

    test('Callback', () => {
        const callback = jest.fn()
        const component = mount(
            <ComponentWithCallback callback={callback}/>
        )

        expect(component).toMatchSnapshot()

        expect(callback.mock.calls.length).toEqual(1)
    })

    test('Updated props', () => {
        const component = mount(
            <ComponentWithHook hasUpdated={false} />
        )

        expect(component).toMatchSnapshot()

        component.setProps({ hasUpdated: true })
        component.update()

        expect(component).toMatchSnapshot()
    })

})
