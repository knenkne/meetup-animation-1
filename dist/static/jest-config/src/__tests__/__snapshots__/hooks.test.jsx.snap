// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hook tests Callback 1`] = `
<ComponentWithCallback
  callback={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div>
    nodejs
  </div>
</ComponentWithCallback>
`;

exports[`Hook tests Updated props 1`] = `
<ComponentWithHook
  hasUpdated={false}
>
  <div>
    Initial Render
  </div>
</ComponentWithHook>
`;

exports[`Hook tests Updated props 2`] = `
<ComponentWithHook
  hasUpdated={true}
>
  <div>
    Updated Name
  </div>
</ComponentWithHook>
`;
